ntp-proto/src/algorithm/kalman/peer.rs

```diff
@@ -421,7 +421,29 @@ impl PeerState {
                     false
                 }
             }
-            PeerStateInner::Stable(filter) => filter.update(config, algo_config, measurement),
+            PeerStateInner::Stable(filter) => {
+                // We check that the difference between the localtime and monotonic
+                // times of the measurement is in line with what would be expected
+                // from recent steering. This check needs to be done here since we
+                // need to revert back to the initial state.
+                if ((measurement.localtime - filter.last_measurement.localtime)
+                    - measurement
+                        .monotime
+                        .abs_diff(filter.last_measurement.monotime))
+                .abs()
+                    > algo_config.meddling_threshold
```

Comment:
can this be broken up? e.g. give some good names to the intermediate results. This looks really weird and is hard to read

Reply:
done.